app/src/androidTest/java/org/mozilla/fenix/ui/robots/TabDrawerRobot.kt

```diff
@@ -229,28 +230,28 @@ class TabDrawerRobot {
         tabMediaControlButton().click()
     }
 
-    private fun clickSelectTabs() {
+    fun clickSelectTabsOption() {
         threeDotMenu().click()
 
-        mDevice.waitNotNull(
-            findObject(
-                text(getStringResource(R.string.tabs_tray_select_tabs))
-            ),
-            waitingTime
-        )
-
-        val selectTabsButton = mDevice.findObject(text(getStringResource(R.string.tabs_tray_select_tabs)))
+        val selectTabsButton = mDevice.findObject(UiSelector().text("Select tabs"))
+        selectTabsButton.waitForExists(waitingTime)
         selectTabsButton.click()
     }
 
-    fun selectTab(title: String) {
-        mDevice.waitNotNull(
-            findObject(text(title)),
-            waitingTime
+    fun selectTab(title: String, tabsSelected: Int) {
+        val tab = mDevice.findObject(
+            UiSelector()
+                .resourceId("$packageName:id/tab_item")
+                .childSelector(UiSelector().text(title))
         )
-
-        val tab = mDevice.findObject(text(title))
+        tab.waitForExists(waitingTime)
         tab.click()
+        try {
+            verifyTabsMultiSelectionCounter(tabsSelected)
+        } catch (e: AssertionError) {
+            tab.click()
+            verifyTabsMultiSelectionCounter(tabsSelected)
+        }
```

Comment:
Do you think a currentTries loop is better here?

Reply:
done